### PR TITLE
fix: changes made after refactoring, validation fixed, error handling improved

### PR DESCRIPTION
**Pull request by Ilya-m2195**

### CODE REVIEW

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      <Title mb={MARGIN_16} order={2}>
+        {t('clientEditing')}
+      </Title>
+      <form
+        onSubmit={form.onSubmit((values) => {
+          if (!id) {
+            return;
+          }
+          dispatch(updateUserThank({ id, values }));
+          form.reset();
```
Comment (AdammmVV) [inline src/components/layout/UserEditing/UserEditing.tsx]: take logic out of markup

File: src/components/shared/AddUserNickForm/AddUserNickForm.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+              id,
+              role: values.role,
+              nickname: values.nickname,
+              balance: values.balance,
+              phone: values.phone,
+              fullName: values.fullName,
+            }),
+          );
+          form.reset();
+        })}
```
Comment (AdammmVV) [inline src/components/shared/AddUserNickForm/AddUserNickForm.tsx]: take logic out of markup

File: src/components/shared/AddUsersForm/AddUsersForm.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  });
+  const inputRef = useMask({ mask: '+7 (___) ___-__-__', replacement: { _: /\d/ } });
+
+  return (
+    <Box maw={700} mb={MARGIN_32} mt={MARGIN_16}>
+      <form
+        onSubmit={form.onSubmit((values) => {
+          console.log(values);
+          form.reset();
+        })}
```
Comment (AdammmVV) [inline src/components/shared/AddUsersForm/AddUsersForm.tsx]: take logic out of markup

File: src/components/shared/AddUsersForm/AddUsersForm.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      >
+        <Group mb={MARGIN_32}>
+          <span>
+            {t('totalCount')} {usersCount}
+          </span>
```
Comment (AdammmVV) [inline src/components/shared/AddUsersForm/AddUsersForm.tsx]: use components from mantine

File: src/components/shared/ErrorMessage/ErrorMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import { useTranslation } from 'react-i18next';
+
+export const ErrorMessage: FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <div className='errorMessageBlock'>
+      <h2>{t('errorMessage')}</h2>
+    </div>
```
Comment (AdammmVV) [inline src/components/shared/ErrorMessage/ErrorMessage.tsx]: use components from mantine

File: src/components/shared/SwitchButton/SwitchButton.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+const sizeValue = 20;
+const blueVale = 6;
+const yellowValue = 4;
+
+export const SwitchButton: FC = () => {
+  const { colorScheme, setColorScheme } = useMantineColorScheme();
+  const theme = useMantineTheme();
+
+  const light = 'light';
+  const dark = 'dark';
```
Comment (AdammmVV) [inline src/components/shared/SwitchButton/SwitchButton.tsx]: put this in config and create Enum

File: src/components/shared/SwitchButton/SwitchButton.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { FC } from 'react';
+
+import { Switch, useMantineTheme, rem, useMantineColorScheme } from '@mantine/core';
+import { IconSun, IconMoonStars } from '@tabler/icons-react';
+
+const sizeValue = 20;
+const blueVale = 6;
+const yellowValue = 4;
```
Comment (AdammmVV) [inline src/components/shared/SwitchButton/SwitchButton.tsx]: put this in constants

File: src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { ChangeEvent, FC, useState } from 'react';
+
+import { NativeSelect } from '@mantine/core';
+import { useTranslation } from 'react-i18next';
+
+export const SwitchLanguageBtn: FC = () => {
+  const en = 'en';
+  const ru = 'ru';
```
Comment (AdammmVV) [inline src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx]: put this in config and create Enum

File: src/constants/constants.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+export const MARGIN_16 = 16;
+export const MARGIN_32 = 32;
+export const pathHome = '/';
+export const pathUsers = '/users';
+export const pathLogin = '/login';
+export const pathSetNick = '/login/setNick';
+export const pathUserEditing = '/users/editing/';
+export const admin = 'admin';
+export const errorMessageNicknameField = 'Данный ник уже занят, попробуйте ввести другой';
+export const errorMessageEmptyField = 'Заполните поле';
```
Comment (AdammmVV) [inline src/constants/constants.ts:8]: put it in translate

File: src/hooks/useSortableData.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  items: Array<any>,
+  config: IConfig | null = null,
+): IReturnType => {
+  const [sortConfig, setSortConfig] = useState<IConfig | null>(config);
+
+  const sortedItems = useMemo(() => {
+    const sortableItems = [...items];
+
+    if (sortConfig !== null) {
+      sortableItems.sort((a, b) => {
```
Comment (AdammmVV) [inline src/hooks/useSortableData.ts]: use clear names

File: src/hooks/useSortableData.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        if (a[sortConfig.key] < b[sortConfig.key]) {
+          return sortConfig.direction === 'ascending' ? -1 : 1;
```
Comment (AdammmVV) [inline src/hooks/useSortableData.ts:34]: use Enum for `ascending, descending`

File: src/hooks/useSortableData.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  direction: string;
+}
+
+interface IReturnType {
+  items: Array<any>;
+  requestSort: (key: string) => void;
+}
+
+export const useSortableData = (
+  items: Array<any>,
```
Comment (AdammmVV) [inline src/hooks/useSortableData.ts]: don`t use any

File: src/hooks/useSortableData.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+  config: IConfig | null = null,
```
Comment (AdammmVV) [inline src/hooks/useSortableData.ts]: what is this for? `null = null`

File: src/hooks/useSortableData.ts [hunk 14]
```diff
@@ -0,0 +1,48 @@
+import { useState, useMemo } from 'react';
+
+interface IConfig {
+  key: string;
+  direction: string;
+}
+
+interface IReturnType {
+  items: Array<any>;
```
Comment (AdammmVV) [inline src/hooks/useSortableData.ts]: don`t use any

File: src/i18n.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import i18n from 'i18next';
+import backend from 'i18next-http-backend';
+import { initReactI18next } from 'react-i18next';
+
+i18n
+  .use(initReactI18next)
+  .use(backend)
+  .init({
+    lng: 'en',
+    fallbackLng: 'en',
```
Comment (AdammmVV) [inline src/i18n.ts]: se Enum and reuse `'en'`

File: src/store/slices/mainSlice.ts [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+  nickname: '',
+  phone: '',
+  balance: 0,
+  isError: false,
+  users: [],
+};
+
+export const logOutUserThank = createAsyncThunk<void, undefined, ThunkApiConfig>(
+  'mainSlice/logOutUser',
+  async (_, config) => {
```
Comment (AdammmVV) [inline src/store/slices/mainSlice.ts]: if you are using ThunkApi, try using destructuring `config`

File: src/components/shared/Navigation/Navigation.tsx [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+
+export const Navigation: FC = () => {
+  const [currentLink, setCurrentLink] = useState(pathHome);
+  const CurrentUserRole = useAppSelector((state) => state.main.role);
+  const { t } = useTranslation();
+
+  const setCurrentLinkHandler = (path: string): void => setCurrentLink(path);
+
+  const activeStyle = 'active';
+  const itemStyle = 'item';
```
Comment (AdammmVV) [inline src/components/shared/Navigation/Navigation.tsx]: use Enum for this

File: src/components/shared/LoaderFC/LoaderFC.tsx [hunk 18]
```diff
@@ -0,0 +1,8 @@
+import { FC } from 'react';
+import '../../../styles/loader.css';
+
+import { Loader } from '@mantine/core';
+
+export const LoaderFC: FC = () => {
+  return <Loader className='loader' color='indigo' />;
```
Comment (AdammmVV) [inline src/components/shared/LoaderFC/LoaderFC.tsx]: use module.css `className='loader'`

File: src/components/shared/ErrorMessage/ErrorMessage.tsx [hunk 19]
```diff
@@ -0,0 +1,14 @@
+import { FC } from 'react';
+import '../../../styles/errorMessage.css';
+
+import { useTranslation } from 'react-i18next';
+
+export const ErrorMessage: FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <div className='errorMessageBlock'>
```
Comment (AdammmVV) [inline src/components/shared/ErrorMessage/ErrorMessage.tsx]: use module.css `className='errorMessageBlock'` and use mantine components

File: src/components/layout/Users/Users.tsx [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+export const Users: FC = () => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(getUsersThank());
+  }, []);
+
+  return (
+    <div>
```
Comment (AdammmVV) [inline src/components/layout/Users/Users.tsx]: use mantine components

File: src/components/layout/Home/Home.tsx [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+
+  return (
+    <div>
+      {email && (
+        <div>
+          <span> {t('welcome')}</span>
+          <span> {nickname}</span>
+        </div>
+      )}
+    </div>
```
Comment (AdammmVV) [inline src/components/layout/Home/Home.tsx]: use mantine components

File: src/App.tsx [hunk 22]
```diff
@@ -1,7 +1,77 @@
...
+            <Route path={pathSetNick} element={<AddUserNickForm />} />
+          </Route>
+          <Route element={<PrivateRoute isAuth={isAuth} />}>
+            <Route path={pathHome} element={<Home />} />
+          </Route>
+          <Route element={<PrivateRouteAdmin />}>
+            <Route path={pathUsers} element={<Users />} />
+            <Route path={`${pathUserEditing}:id`} element={<UserEditing />} />
+          </Route>
+        </Routes>
```
Comment (AdammmVV) [inline src/App.tsx]: try use [createBrowserRouter](https://reactrouter.com/en/main/start/overview#client-side-routing)

File: package.json [hunk 23]
```diff
@@ -3,15 +3,28 @@
   "version": "0.1.0",
```
Comment (AdammmVV) [inline package.json:3]: why didn't use Vite React instead of CRA?

File: src/App.tsx [hunk 24]
```diff
@@ -27,6 +33,7 @@ import {
   pathUsers,
 } from './constants/constants';
 import { useAppSelector } from './hooks/useAppSelector';
+import React from 'react';
 
 export const App: FC = () => {
   const isLoading = useAppSelector((state) => state.main.isLoading);
```
Comment (suveiko) [inline src/App.tsx]: create separate file for selectors, for example: 
`
const getIsLoading = (state: RootState): boolean => state.main.isLoading

usage: 
const isLoading = useAppSelector(getIsLoading)
`

File: src/components/layout/Home/Home.tsx [hunk 25]
```diff
@@ -1,22 +1,24 @@
...
 
+import { Box, Group, Text } from '@mantine/core';
 import { useTranslation } from 'react-i18next';
 
 import { useAppSelector } from '../../../hooks/useAppSelector';
+import React from 'react';
 
 export const Home: FC = () => {
   const nickname = useAppSelector((state) => state.main.nickname);
   const email = useAppSelector((state) => state.main.email);
```
Comment (suveiko) [inline src/components/layout/Home/Home.tsx]: also move to selectors

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 26]
```diff
@@ -41,15 +56,7 @@ export const UserEditing: FC = () => {
...
-      <form
-        onSubmit={form.onSubmit((values) => {
-          if (!id) {
-            return;
-          }
-          dispatch(updateUserThank({ id, values }));
-          form.reset();
-        })}
-      >
+      <form onSubmit={form.onSubmit((values) => onSubmitHandler(values))}>
```
Comment (suveiko) [inline src/components/layout/UserEditing/UserEditing.tsx]: `form.onSubmit(onSubmitHandler)`

File: src/components/shared/AddUserNickForm/AddUserNickForm.tsx [hunk 27]
```diff
@@ -20,6 +29,21 @@ export const AddUserNickForm: FC = () => {
...
+  const onSubmitHandler = (values: IValues): void => {
+    dispatch(
+      addUserThank({
+        email,
+        id,
+        role: values.role,
+        nickname: values.nickname,
+        balance: values.balance,
+        phone: values.phone,
+        fullName: values.fullName,
```
Comment (suveiko) [inline src/components/shared/AddUserNickForm/AddUserNickForm.tsx:41]: `      role,
        nickname,
        balance,
        phone,
        fullName,
`

File: src/components/shared/AddUserNickForm/AddUserNickForm.tsx [hunk 28]
```diff
@@ -20,6 +29,21 @@ export const AddUserNickForm: FC = () => {
     navigate(-1);
   };
 
+  const onSubmitHandler = (values: IValues): void => {
```
Comment (suveiko) [inline src/components/shared/AddUserNickForm/AddUserNickForm.tsx]: ({ role, nickname, balance, phone, fullName }: IValues)

File: src/components/shared/AddUserNickForm/AddUserNickForm.tsx [hunk 29]
```diff
@@ -42,22 +66,7 @@ export const AddUserNickForm: FC = () => {
...
-              nickname: values.nickname,
-              balance: values.balance,
-              phone: values.phone,
-              fullName: values.fullName,
-            }),
-          );
-          form.reset();
-        })}
-      >
+      <form onSubmit={form.onSubmit((values) => onSubmitHandler(values))}>
```
Comment (suveiko) [inline src/components/shared/AddUserNickForm/AddUserNickForm.tsx]: `<form onSubmit={form.onSubmit(onSubmitHandler)}>`

File: src/components/shared/AddUsersForm/AddUsersForm.tsx [hunk 30]
```diff
@@ -23,20 +32,19 @@ export const AddUsersForm: FC<Props> = ({ usersCount }) => {
       balance: 0,
     },
   });
+  const onSubmitHandler = (values: IValues): void => {
+    console.log(values);
```
Comment (suveiko) [inline src/components/shared/AddUsersForm/AddUsersForm.tsx:29]: remove log

File: src/components/shared/AddUsersForm/AddUsersForm.tsx [hunk 31]
```diff
@@ -23,20 +32,19 @@ export const AddUsersForm: FC<Props> = ({ usersCount }) => {
...
 
   return (
     <Box maw={700} mb={MARGIN_32} mt={MARGIN_16}>
-      <form
-        onSubmit={form.onSubmit((values) => {
-          console.log(values);
-          form.reset();
-        })}
-      >
+      <form onSubmit={form.onSubmit((values) => onSubmitHandler(values))}>
```
Comment (suveiko) [inline src/components/shared/AddUsersForm/AddUsersForm.tsx]: `<form onSubmit={form.onSubmit(onSubmitHandler)}>`

File: src/components/shared/Navigation/Navigation.tsx [hunk 32]
```diff
@@ -16,22 +18,27 @@ export const Navigation: FC = () => {
...
   return (
     <Group justify='center' mt={-30}>
       <Link
         to={pathHome}
-        className={currentLink === pathHome ? activeStyle : itemStyle}
+        className={
+          currentLink === pathHome
+            ? NamesActiveStyles.activeStyle
+            : NamesActiveStyles.itemStyle
+        }
```
Comment (suveiko) [inline src/components/shared/Navigation/Navigation.tsx]: move to variable

File: src/components/shared/Navigation/Navigation.tsx [hunk 33]
```diff
@@ -16,22 +18,27 @@ export const Navigation: FC = () => {
...
         onClick={() => setCurrentLinkHandler(pathHome)}
```
Comment (suveiko) [inline src/components/shared/Navigation/Navigation.tsx]: move outside to separate function

File: src/components/shared/Navigation/Navigation.tsx [hunk 34]
```diff
@@ -16,22 +18,27 @@ export const Navigation: FC = () => {
...
         {t('home')}
       </Link>
       {CurrentUserRole === admin && (
         <Link
           to={pathUsers}
-          className={currentLink === pathUsers ? activeStyle : itemStyle}
+          className={
+            currentLink === pathUsers
+              ? NamesActiveStyles.activeStyle
+              : NamesActiveStyles.itemStyle
```
Comment (suveiko) [inline src/components/shared/Navigation/Navigation.tsx]: move to variable

File: src/components/shared/Navigation/Navigation.tsx [hunk 35]
```diff
@@ -16,22 +18,27 @@ export const Navigation: FC = () => {
...
+          }
           onClick={() => setCurrentLinkHandler(pathUsers)}
```
Comment (suveiko) [inline src/components/shared/Navigation/Navigation.tsx]: move to separate function

File: src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx [hunk 36]
```diff
@@ -19,7 +20,7 @@ export const SwitchLanguageBtn: FC = () => {
       w={75}
       value={value}
       onChange={(e: ChangeEvent<HTMLSelectElement>) => changeLanguageHandler(e)}
```
Comment (suveiko) [inline src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx]: `onChange={changeLanguageHandler}`

File: package.json [hunk 37]
```diff
@@ -3,24 +3,40 @@
...
+    "i18next": "^23.7.16",
+    "i18next-http-backend": "^2.4.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-scripts": "5.0.1",
+    "react-i18next": "^14.0.0",
+    "react-redux": "^9.0.4",
+    "react-router-dom": "^6.21.1",
     "typescript": "^4.4.2",
+    "vite": "^5.0.12",
```
Comment (remadyt) [inline package.json:28]: i think its' dev depend

File: package.json [hunk 38]
```diff
@@ -3,24 +3,40 @@
...
+    "@reduxjs/toolkit": "^2.0.1",
+    "@tabler/icons-react": "^2.45.0",
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^13.0.0",
     "@testing-library/user-event": "^13.2.1",
     "@types/jest": "^27.0.1",
     "@types/node": "^16.7.13",
     "@types/react": "^18.0.0",
     "@types/react-dom": "^18.0.0",
+    "@vitejs/plugin-react": "^4.2.1",
```
Comment (remadyt) [inline package.json:19]: i think its' dev depend

File: package.json [hunk 39]
```diff
@@ -3,24 +3,40 @@
...
+    "i18next-http-backend": "^2.4.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-scripts": "5.0.1",
+    "react-i18next": "^14.0.0",
+    "react-redux": "^9.0.4",
+    "react-router-dom": "^6.21.1",
     "typescript": "^4.4.2",
+    "vite": "^5.0.12",
+    "vite-tsconfig-paths": "^4.3.1",
```
Comment (remadyt) [inline package.json:29]: i think its' dev depend

File: package.json [hunk 40]
```diff
@@ -3,24 +3,40 @@
...
     "web-vitals": "^2.1.0"
```
Comment (remadyt) [inline package.json:30]: this not needed with vite

File: src/App.tsx [hunk 41]
```diff
@@ -1,7 +1,85 @@
...
+import './styles/reset.css';
+
+import { Home } from './components/layout/Home/Home';
+import { LoginPage } from './components/layout/LoginPage/LoginPage';
+import { UserEditing } from './components/layout/UserEditing/UserEditing';
+import { Users } from './components/layout/Users/Users';
+import { AddUserNickForm } from './components/shared/AddUserNickForm/AddUserNickForm';
+import { LogOutUser } from './components/shared/auth/LogOutUser';
+import { ErrorMessage } from './components/shared/ErrorMessage/ErrorMessage';
+import { LoaderFC } from './components/shared/LoaderFC/LoaderFC';
```
Comment (remadyt) [inline src/App.tsx]: why need FC in namin?

File: src/api/api.ts [hunk 42]
```diff
@@ -0,0 +1,85 @@
...
+  nameCollection: string,
+  id: string,
+): Promise<DocumentData | void> => {
+  const docRef = doc(dataBase, nameCollection, id);
+  const docInfo = await getDoc(docRef);
+
+  return docInfo.data();
+};
+
+export const LogInUser = async (
```
Comment (remadyt) [inline src/api/api.ts]: why big first letter?

File: src/components/layout/LoginPage/LoginPage.tsx [hunk 43]
```diff
@@ -0,0 +1,17 @@
+import { FC } from 'react';
+
+import { Group } from '@mantine/core';
+
+import { SignInGitHub } from '../../shared/auth/SignInGitHub';
+import { SignInGoogle } from '../../shared/auth/SignInGoogle';
+import React from 'react';
+
+export const LoginPage: FC = () => {
+
```
Comment (remadyt) [inline src/components/layout/LoginPage/LoginPage.tsx]: remove empty line

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 44]
```diff
@@ -0,0 +1,83 @@
...
+export const UserEditing: FC = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const users = useAppSelector(getUsers);
+  const { t } = useTranslation();
+  const { id } = useParams();
+
+  const goBackHandler = (): void => {
+    navigate(-1);
+  };
```
Comment (remadyt) [inline src/components/layout/UserEditing/UserEditing.tsx:25]: add empty line after

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 45]
```diff
@@ -0,0 +1,83 @@
...
+
+interface IValues {
+  nickname: string;
+  fullName: string;
+  role: string;
+}
+
+export const UserEditing: FC = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
```
Comment (remadyt) [inline src/components/layout/UserEditing/UserEditing.tsx:17]: separate hooks by logic, here and all such places

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 46]
```diff
@@ -0,0 +1,83 @@
...
+
+import { MARGIN_16 } from '../../../constants/constants';
+import { validateNickname } from '../../../helpers/validateNickname';
+import { useAppDispatch } from '../../../hooks/useAppDispatch';
+import { useAppSelector } from '../../../hooks/useAppSelector';
+import { updateUserThank } from '../../../store/slices/mainSlice';
+import React from 'react';
+import { getUsers } from '../../../store/selectors';
+
+interface IValues {
```
Comment (remadyt) [inline src/components/layout/UserEditing/UserEditing.tsx]: move to types.ts here and all such places. only props in component file

File: src/components/layout/UserEditing/UserEditing.tsx [hunk 47]
```diff
@@ -0,0 +1,83 @@
...
+    if (!id) {
+      return;
+    }
+    dispatch(updateUserThank({ id, values }));
+    form.reset();
+  };
+
+  const onSubmitHandler = (values: IValues) => onSubmitValues(values);
+
+  const form = useForm({
```
Comment (remadyt) [inline src/components/layout/UserEditing/UserEditing.tsx:35]: useForm<IValues>

File: src/components/layout/Users/Users.tsx [hunk 48]
```diff
@@ -0,0 +1,25 @@
+import { FC, useEffect } from 'react';
+
+import { Group, Title } from '@mantine/core';
+import { useTranslation } from 'react-i18next';
+
+import { useAppDispatch } from '../../../hooks/useAppDispatch';
```
Comment (remadyt) [inline src/components/layout/Users/Users.tsx]: use absolute paths, need put in tsconfig baseUrl: 'src'

File: src/components/shared/AddUsersForm/AddUsersForm.tsx [hunk 49]
```diff
@@ -0,0 +1,80 @@
...
+        <Group mb={MARGIN_32}>
+          <Text>
+            {t('totalCount')} {usersCount}
+          </Text>
+          <Button type='submit'>{t('addClient')}</Button>
+        </Group>
+        <Flex gap={MARGIN_16}>
+          <TextInput
+            ref={inputRef}
+            mb={MARGIN_16}
```
Comment (remadyt) [inline src/components/shared/AddUsersForm/AddUsersForm.tsx]: for margin const not needed, use variable, 'xs', 'm'

File: src/components/shared/ButtonSort/ButtonSort.tsx [hunk 50]
```diff
@@ -0,0 +1,22 @@
+import { FC } from 'react';
+
+import { Button } from '@mantine/core';
+import { IconArrowsSort } from '@tabler/icons-react';
+
+import { UsersKeys } from '../../../constants/enums';
+import React from 'react';
```
Comment (remadyt) [inline src/components/shared/ButtonSort/ButtonSort.tsx]: react import not needed

File: src/components/shared/ButtonSort/ButtonSort.tsx [hunk 51]
```diff
@@ -0,0 +1,22 @@
...
+
+type Props = {
+  nameField: UsersKeys;
+  requestSort: (fill: UsersKeys) => void;
+};
+
+export const ButtonSort: FC<Props> = ({ nameField, requestSort }) => {
+  const requestSortHandler = (): void => requestSort(nameField);
```
Comment (remadyt) [inline src/components/shared/ButtonSort/ButtonSort.tsx]: add {}

File: src/components/shared/LoaderFC/LoaderFC.tsx [hunk 52]
```diff
@@ -0,0 +1,10 @@
+import { FC } from 'react';
+
+import { Loader } from '@mantine/core';
+
+import style from './loaderFC.module.css';
+import React from 'react';
```
Comment (remadyt) [inline src/components/shared/LoaderFC/LoaderFC.tsx]: same, remove here and all such places

File: src/components/shared/Navigation/Navigation.tsx [hunk 53]
```diff
@@ -0,0 +1,53 @@
...
+import { useAppSelector } from '../../../hooks/useAppSelector';
+import React from 'react';
+import { getRole } from '../../../store/selectors';
+
+export const Navigation: FC = () => {
+  const [currentLink, setCurrentLink] = useState(pathHome);
+  const currentUserRole = useAppSelector(getRole);
+  const { t } = useTranslation();
+
+  const currentClassName =
```
Comment (remadyt) [inline src/components/shared/Navigation/Navigation.tsx]: use clsx library

File: src/components/shared/Navigation/Navigation.tsx [hunk 54]
```diff
@@ -0,0 +1,53 @@
...
+        className={currentClassName}
+        onClick={setCurrentLinkHandler}
+      >
+        {t('home')}
+      </Link>
+      {currentUserRole === admin && (
+        <Link
+          to={pathUsers}
+          className={
+            currentLink === pathUsers
```
Comment (remadyt) [inline src/components/shared/Navigation/Navigation.tsx]: use isAcitve in NavLink component

File: src/components/shared/SwitchButton/SwitchButton.tsx [hunk 55]
```diff
@@ -0,0 +1,41 @@
...
+
+import { sizeValue, yellowValue, blueVale } from './config';
+import React from 'react';
+
+export const SwitchButton: FC = () => {
+  const { colorScheme, setColorScheme } = useMantineColorScheme();
+  const theme = useMantineTheme();
+
+  const setLight = (): void => setColorScheme(ColorTheme.light);
+  const setDark = (): void => setColorScheme(ColorTheme.dark);
```
Comment (remadyt) [inline src/components/shared/SwitchButton/SwitchButton.tsx]: refactor with 1 function

File: src/components/shared/SwitchButton/SwitchButton.tsx [hunk 56]
```diff
@@ -0,0 +1,41 @@
...
+
+  const sunIcon = (
+    <IconSun
+      style={{ width: rem(sizeValue), height: rem(sizeValue) }}
```
Comment (remadyt) [inline src/components/shared/SwitchButton/SwitchButton.tsx]: do not use inline styles

File: src/store/slices/mainSlice.ts [hunk 57]
```diff
@@ -0,0 +1,178 @@
...
+  role: '',
+  nickname: '',
+  phone: '',
+  balance: 0,
+  isError: false,
+  users: [],
+};
+
+export const logOutUserThank = createAsyncThunk<void, undefined, ThunkApiConfig>(
+  'mainSlice/logOutUser',
```
Comment (remadyt) [inline src/store/slices/mainSlice.ts:41]: why need prefix main Slice?

File: src/store/slices/mainSlice.ts [hunk 58]
```diff
@@ -0,0 +1,178 @@
...
+      return rejectWithValue(error);
+    }
+  },
+);
+
+export const getUsersThank = createAsyncThunk<Array<IUser>, void, ThunkApiConfig>(
+  'mainSlice/getUsersThank',
+  async (_, { rejectWithValue }) => {
+    try {
+      const result = await getUsers();
```
Comment (remadyt) [inline src/store/slices/mainSlice.ts:77]: get all users for check nickname it's bad, why u not use checkValueExist?

File: src/store/slices/mainSlice.ts [hunk 59]
```diff
@@ -0,0 +1,178 @@
...
+);
+
+export const addUserThank = createAsyncThunk<void, IUser, ThunkApiConfig>(
+  'mainSlice/addUserThank',
+  async (value, { dispatch, rejectWithValue }) => {
+    try {
+      await addUser(value);
+      dispatch(setUser(value));
+    } catch (error) {
+      return rejectWithValue(error);
```
Comment (remadyt) [inline src/store/slices/mainSlice.ts]: add errorHandler to all catch block

File: src/store/store.ts [hunk 60]
```diff
@@ -0,0 +1,12 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+import mainReducer from './slices/mainSlice';
+
+export const store = configureStore({
+  reducer: {
+    main: mainReducer,
```
Comment (remadyt) [inline src/store/store.ts:7]: use more informative name

File: yarn.lock
Comment (remadyt) [inline yarn.lock:1]: remove yarn lock. use npm or yarn, not together

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/store/slices/mainSlice.ts [hunk 61]
```diff
@@ -0,0 +1,223 @@
...
+    },
+    setRole: (state, action) => {
+      state.role = action.payload.role;
+    },
+    setIsOccupiedNick: (state, action: PayloadAction<boolean>) => {
+      state.isOccupiedNick = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(logOutUserThank.pending, (state) => {
```
Comment (AdammmVV) [inline src/store/slices/mainSlice.ts:150]: try using [addMatcher](https://redux-toolkit.js.org/api/createreducer#builderaddmatcher) for the same thunk

File: src/store/slices/mainSlice.ts [hunk 62]
```diff
@@ -0,0 +1,223 @@
...
+      state.id = '';
+      state.role = '';
+      state.users = [];
+      state.isLoading = false;
+      state.isError = false;
+    });
+    builder.addCase(logOutUserThank.rejected, (state) => {
+      state.isLoading = false;
+      state.isError = true;
+    });
```
Comment (AdammmVV) [inline src/store/slices/mainSlice.ts]: why don't you handle errors from rejectWithValue(error)

File: src/components/shared/auth/LogOutUser.tsx [hunk 63]
```diff
@@ -0,0 +1,22 @@
...
+
+export const LogOutUser: FC = () => {
+  const dispatch = useAppDispatch();
+  
+  const logOutHandler = (): void => {
+    dispatch(logOutUserThank());
+  };
+
+  return (
+    <Button variant='outline' color='rgba(99, 99, 99, 1)' onClick={logOutHandler}>
```
Comment (AdammmVV) [inline src/components/shared/auth/LogOutUser.tsx]: use color from theme

File: src/components/shared/ErrorMessage/ErrorMessage.tsx [hunk 64]
```diff
@@ -0,0 +1,17 @@
...
+import styles from './errorMessage.module.css';
+import React from 'react';
+
+export const ErrorMessage: FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <Title mt={32} className={styles.errorMessage} order={2}>
+      {t('errorMessage')}
+    </Title>
```
Comment (AdammmVV) [inline src/components/shared/ErrorMessage/ErrorMessage.tsx]: you need to make sure that the error text is displayed and not a stub

File: src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx [hunk 65]
```diff
@@ -0,0 +1,26 @@
...
+import React from 'react';
+
+export const SwitchLanguageBtn: FC = () => {
+  const [value, setValue] = useState<Languages | string>(Languages.en);
+  const { i18n } = useTranslation();
+
+  const changeLanguageHandler = (e: ChangeEvent<HTMLSelectElement>): void => {
+    setValue(e.currentTarget.value);
+    i18n.changeLanguage(e.currentTarget.value);
+  };
```
Comment (AdammmVV) [inline src/components/shared/SwitchLanguageBtn/SwitchLanguageBtn.tsx]: need to save to local storage

File: src/components/shared/UsersTable/UsersTable.tsx [hunk 66]
```diff
@@ -0,0 +1,71 @@
...
+import { UserName } from '../UserName/UserName';
+import React from 'react';
+import { getUsers } from 'store/selectors';
+
+export const UsersTable: FC = () => {
+  const users = useAppSelector(getUsers);
+  const { sortUsers, requestSort } = useSortableData(users);
+  const { t } = useTranslation();
+
+  const rows = sortUsers.map((element) => (
```
Comment (AdammmVV) [inline src/components/shared/UsersTable/UsersTable.tsx]: use more informative name for `element`

File: src/App.tsx [hunk 67]
```diff
@@ -1,7 +1,78 @@
...
+            <Route path={pathLogin} element={<LoginPage />} />
+            <Route path={pathSetNick} element={<AddUserNickForm />} />
+          </Route>
+          <Route element={<PrivateRoute />}>
+            <Route path={pathHome} element={<Home />} />
+          </Route>
+          <Route element={<PrivateRouteAdmin />}>
+            <Route path={pathUsers} element={<Users />} />
+            <Route path={`${pathUserEditing}:id`} element={<UserEditing />} />
+          </Route>
```
Comment (AdammmVV) [inline src/App.tsx:85]: add rout for error hendle in rout `error 404`

**Pull request merged**